q2_demux/_demux.py

```diff
@@ -48,6 +52,47 @@ def __next__(self):
         return next(self.barcode_generator), next(self.sequence_generator)
 
 
+def summary(output_dir: str, data: SingleLanePerSampleSingleEndFastqDirFmt) \
+        -> None:
+    per_sample_fastqs = list(data.sequences.iter_views(FastqGzFormat))
+    per_sample_fastq_counts = {}
+    for per_sample_fastq in per_sample_fastqs:
+        seqs = skbio.io.read(per_sample_fastq[1].open(),
+                             format='fastq',
+                             phred_offset=33, compression='gzip',
+                             constructor=skbio.DNA)
+        count = 0
+        for seq in seqs:
+            count += 1
+        per_sample_fastq_counts[per_sample_fastq[0]] = count
+
+    result = pd.Series(per_sample_fastq_counts)
+    result.sort_values(inplace=True, ascending=False)
+    result.to_csv(os.path.join(output_dir,
+                  'per-sample-fastq-counts.csv'))
+    ax = sns.distplot(result, kde=False)
+    ax.set_xlabel('Number of sequences')
+    ax.set_xlabel('Frequency')
+    fig = ax.get_figure()
+    fig.savefig(os.path.join(output_dir, 'demultiplex-summary.png'))
+    fig.savefig(os.path.join(output_dir, 'demultiplex-summary.pdf'))
+    with open(os.path.join(output_dir, 'index.html'), 'w') as fh:
+        fh.write('<html><body>\n')
+        fh.write(' <h1>Demultiplexed sequence counts summary</h1>\n')
+        fh.write(' Minimum: %d<br>\n' % result.min())
+        fh.write(' Median: %d<br>\n' % result.max())
+        fh.write(' Mean: %1.3f<br>\n' % result.mean())
+        fh.write(' Maximum: %1.3f<br>\n' % result.median())
+        fh.write(' Total number of sequences demultiplexed: %d<p>\n\n' %
+                 result.sum())
+        fh.write(' <img src="demultiplex-summary.png"><br>\n')
+        fh.write(' <a href="demultiplex-summary.pdf">PDF</a><p>\n\n')
+        fh.write(' <h1>Demultiplexed sequence counts detail</h1>\n')
```

Comment:
We think this header makes more sense, given the series column names:

"Per-sample sequence counts"